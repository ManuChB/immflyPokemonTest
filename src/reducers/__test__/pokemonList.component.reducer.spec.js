import pokemonAction from '../../actions/pokemonList.component.action';
import pokemonReducer from '../pokemonList.component.reducer';

describe('PokemonList reducer', () => {
    const initialState = {
        pokemonList: [],
        selectedPokemon: null,
        showInfoModal: false,
        loading: false
    };

    const pokemon = {
        name: 'eevee',
        url: '',
        height: 4,
        types: [],
        abilities: []
    };
    const pokemonList =[pokemon];
    
    it('handle setPokemonList()', () => {
        expect(pokemonReducer(initialState, pokemonAction.setPokemonList(pokemonList))).toMatchSnapshot();
    });

    it('handle setSelectedPokemonInfo()', () => {
        expect(pokemonReducer(initialState, pokemonAction.setSelectedPokemonInfo(pokemon))).toMatchSnapshot();
    });

    it('handle closeModal()', () => {
        expect(pokemonReducer(initialState, pokemonAction.closeModal())).toMatchSnapshot();
    });

    it('handle setLoading()', () => {
        const loading = false;
        expect(pokemonReducer(initialState, pokemonAction.setLoading(loading))).toMatchSnapshot();
    });

});
