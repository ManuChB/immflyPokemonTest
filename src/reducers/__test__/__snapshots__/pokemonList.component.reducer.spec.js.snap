// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList reducer handle closeModal() 1`] = `
Object {
  "loading": false,
  "pokemonList": Array [],
  "selectedPokemon": null,
  "showInfoModal": false,
}
`;

exports[`PokemonList reducer handle setLoading() 1`] = `
Object {
  "loading": false,
  "pokemonList": Array [],
  "selectedPokemon": null,
  "showInfoModal": false,
}
`;

exports[`PokemonList reducer handle setPokemonList() 1`] = `
Object {
  "loading": false,
  "pokemonList": Array [
    Object {
      "abilities": Array [],
      "height": 4,
      "name": "eevee",
      "types": Array [],
      "url": "",
    },
  ],
  "selectedPokemon": null,
  "showInfoModal": false,
}
`;

exports[`PokemonList reducer handle setSelectedPokemonInfo() 1`] = `
Object {
  "loading": false,
  "pokemonList": Array [],
  "selectedPokemon": Object {
    "abilities": Array [],
    "height": 4,
    "name": "eevee",
    "types": Array [],
    "url": "",
  },
  "showInfoModal": true,
}
`;
